### PR TITLE
Allow not supplying some parameters to ApiEditSpace

### PR DESCRIPTION
**Pull request by wgevaert**
fixes [issue 9](https://github.com/Open-CSP/WSSpaces/issues/9)

### CODE REVIEW

File: src/API/ApiEditSpace.php [hunk 1]
```diff
@@ -66,50 +86,12 @@ public function execute() {
...
-					$this->msg( "wss-api-space-does-not-exist", $request_params["nsid"] )->parse()
-				)
-			);
-		}
-
+	private function validateParams (
+		Space $space,
+		?string $ns_key,
+		?string $ns_name,
+		?string $ns_descripotion
```
Comment (marijnvanwezel) [inline src/API/ApiEditSpace.php:93]: `$ns_description`

**Comment by marijnvanwezel** [comment]
Did you test these changes?

**Comment by marijnvanwezel** [comment]
The frontend also requires a namespace name, key and description. Isn't it weird to make them nullable in the API?

**Comment by wgevaert** [comment]
> The frontend also requires a namespace name, key and description. Isn't it weird to make them nullable in the API?

In the frontend, these are auto-filled on page load. Therefore, leaving these empty requires extra effort by the end-user.

In the api, these values are not readily available (see [issue 8](https://github.com/Open-CSP/WSSpaces/issues/8)), and the only way to obtain these currently is by doing a full query to `ApiQuerySpaces` and parsing the returned list of alll the WSS-namespaces. This seems a bit too much effort for something that we can just do like this pull request does.

> Did you test these changes?

I did yes, in the console.
I did some more tests and found that the admins can be nulled by supplying faulty variables. I added some validation for that.

**Comment by wgevaert** [comment]
> In the api, these values are not readily available (see https://github.com/Open-CSP/WSSpaces/issues/8), and the only way to obtain these currently is by doing a full query to ApiQuerySpaces and parsing the returned list of alll the WSS-namespaces.

This seemed a bit stupid to me, so I made [another pull request that addresses that issue](https://github.com/Open-CSP/WSSpaces/pull/12)

**Pull request merged**